docs/airgap-install.md

```diff
@@ -1,69 +1,53 @@
-  through `k0s.yaml`
-- Setup private registry and override image names (not covered by this document)
+## Steps
 
-## Bundle preparation
+#### 1. Create OCI bundle
 
-You need following images in the bundle:
+To build OCI bundle with images we can utilize the fact that containerd pulls all images during the k0s worker normal bootstrap.
+Use following commands on the machine with previously installed k0s worker:
 
 ```
-us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-agent:v0.0.13
-gcr.io/k8s-staging-metrics-server/metrics-server:v0.3.7
-k8s.gcr.io/kube-proxy:v1.20.4
-docker.io/coredns/coredns:1.7.0
-docker.io/calico/cni:v3.16.2
-docker.io/calico/pod2daemon-flexvol:v3.16.2
-docker.io/calico/node:v3.16.2
-docker.io/calico/kube-controllers:v3.16.2
-k8s.gcr.io/pause:3.2
-``` 
+# # The command k0s airgap list-images prints images used by the current setup
+# # It respects the k0s.yaml values
 
-To generate images list you can use k0s command:
-```k0s airgap list-images```
+# export IMAGES=k0s airgap list-images | xargs
+# ctr --namespace k8s.io --address /run/k0s/containerd.sock images export bundle_file $IMAGES 
+```
 
-You need to install containerd cli tool [ctr](https://containerd.io/downloads/)
+Pay attention to the `address` and `namespace` arguments given to the `ctr` tool.
 
-### Default bundle
+#### 2. Sync bundle file with airgapped machine
 
-Manually setup k0s cluster with at least one worker and use the following command to export images from the cluster:
+Provide the `bundle_file` from the previous step to the target machine. Locate the file under the `images` directory in the k0s data directory.
+Use following commands to place bundle into the default location:
 
 ```
-export IMAGES=$(k0s airgap list-images | xargs)
-ctr --namespace k8s.io --address /run/k0s/containerd.sock images export bundle.tgz $IMAGES
+# mkdir -p /var/lib/k0s/images
+# cp bundle_file /var/lib/k0s/images/bundle_file
 ```
 
-### Custom bundle
-
-Custom bundle could be useful in case if you need to pre-install any helm based extensions or apply custom kubernetes
-manifests. Just extend the images list received from k0s command with any custom images.
-
-## Starting up worker with a bundle
-
-During the boot process k0s project will import any "tgz" file found under the `/var/lib/k0s/images` directory.
-
-## Fully offline mode
-
-To ensure that kubelet uses only images from the bundle use corresponding k0s.yaml:
+#### 3. Ensure pull policy in the k0s.yaml (optional)
 
+Use following k0s.yaml to force bundled ContainerD never pull images for the k0s components.
 ```
 apiVersion: k0s.k0sproject.io/v1beta1
 kind: Cluster
-images:
-  default_pull_policy: Never
+metadata:
+  name: k0s
+spec:
+  images:
+    default_pull_policy: Never
 ```
 
-Important notification here is that `default_pull_policy` affects only images installed by the k0s itself, not images
-from the helm extensions or any other manually installed images.
-
-## Private registry
+#### 4. Run worker
```

Comment:
Something could be said about the controller, how the controller is started in the airgap environment. Is the standard k0s binary enough?